Cargo.toml

```diff
@@ -53,11 +53,16 @@ rand = { version = "0.8", optional = true }
 async-std = { version = "1.5.0", optional = true}
 async-trait = "0.1.24"
 
-# Only needed for TLS
+# Only needed for native tls
 native-tls = { version = "0.2", optional = true }
 tokio-native-tls = { version = "0.3", optional = true }
 async-native-tls = { version = "0.4", optional = true }
 
+# Only needed for rustls
+rustls = { version = "0.20.4", optional = true }
+webpki-roots = { version = "0.22.3", optional = true }
```

Comment:
Ideally we would also offer the choice of using rustls-native-certs here instead which I think is preferred in most cases, but feel free to leave that out of this PR.